common/transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/MqttTransportHandler.java

```diff
@@ -366,6 +380,30 @@ private void handleGatewayPublishMsg(ChannelHandlerContext ctx, String topicName
         }
     }
 
+    private void handleSparkplugPublishMsg(ChannelHandlerContext ctx, SparkplugTopic sparkplugTopic, int msgId, MqttPublishMessage mqttMsg) {
```

Comment:
It looks like we should move this part to SparkPlugHandler.

Reply:
I disagree. We should keep the code very similar to the gateway part. It is ok to keep the switch here.

Reply:
Ok